src/main/scala/mesosphere/marathon/core/volume/VolumesModule.scala

```diff
@@ -0,0 +1,54 @@
+package mesosphere.marathon.core.volume
+
+import com.wix.accord.Validator
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.state.AppDefinition
+import mesosphere.marathon.state.Volume
+import mesosphere.marathon.core.volume.impl._
+
+trait LocalVolumes {
+  /** @return a stream of task local volumes, extrapolating them from the app spec */
+  def local(app: AppDefinition): Iterable[Task.LocalVolume]
+  /** @return the aggregate mesos disk resources required for volumes */
+  def diskSize(app: AppDefinition): Double
+}
+
+/**
+  * VolumeProvider is an interface implemented by storage volume providers
+  */
+trait VolumeProvider[+T <: Volume] {
+  /** name uniquely identifies this volume provider */
+  val name: String
+  /** validation implements this provider's specific validation rules */
+  val validation: Validator[Volume]
```

Comment:
we could make this a func that accepts a Container spec, but how to we determine the default containerizer? I don't think the logic for that is centralized...